sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/MediaStreamingPackageParser.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation.models;
+
+import com.azure.communication.callautomation.implementation.models.MediaStreamingAudioInternal;
+import com.azure.communication.callautomation.implementation.models.MediaStreamingMetadataInternal;
+import com.azure.communication.common.CommunicationUserIdentifier;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parser of the different packages received as part of Media streaming.
+ */
+public class MediaStreamingPackageParser {
+    private static final ClientLogger LOGGER = new ClientLogger(MediaStreamingPackageParser.class);
+
+    /***
+     * Parses a Media Streaming package from BinaryData.
+     *
+     * @param json The MediaStreaming package as a BinaryData obejct.
+     * @throws RuntimeException Any exceptions occurs at runtime.
+     * @return a MediaStreamingPackageBase object.
+     */
+    public static MediaStreamingPackageBase parse(BinaryData json) {
+        return parse(json.toString());
+    }
+
+    /***
+     * Parses a Media Streaming package from byte array.
+     *
+     * @param receivedBytes The MediaStreaming package as a byte[].
+     * @throws RuntimeException Any exceptions occurs at runtime.
+     * @return a MediaStreamingPackageBase object.
+     */
+    public static MediaStreamingPackageBase parse(byte[] receivedBytes) {
+        return parse(new String(receivedBytes, StandardCharsets.UTF_8));
+    }
+
+    /***
+     * Parses a Media Streaming package from String.
+     *
+     * @param stringJson The MediaStreaming package as a String.
+     * @throws RuntimeException Any exceptions occurs at runtime.
+     * @return a MediaStreamingPackageBase object.
+     */
+    public static MediaStreamingPackageBase parse(String stringJson) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+            JsonNode jsonData = mapper.readTree(stringJson);
+            if (stringJson.contains("data")) {
+                MediaStreamingAudioInternal audioInternal = mapper.convertValue(jsonData, MediaStreamingAudioInternal.class);
+                return new MediaStreamingAudio(audioInternal.getAudioData(), OffsetDateTime.parse(audioInternal.getTimestamp(), DateTimeFormatter.ISO_OFFSET_DATE_TIME), new CommunicationUserIdentifier(audioInternal.getParticipantRawID()), audioInternal.isSilent());
```

Comment:
I wonder if it is better to let the constructor to handle the data type translations.